src/common/commonutils/CommandUtils.c

```diff
@@ -29,270 +31,200 @@ static int NormalizeStatus(int status)
-
-    maximumCommandLine = (size_t)sysconf(_SC_ARG_MAX);
-    if (commandLineLength > maximumCommandLine)
-    {
-        OsConfigLogDebug(log, "Cannot run command '%s', command too long (%u), ARG_MAX: %u", command, (unsigned)commandLineLength, (unsigned)maximumCommandLine);
-        return E2BIG;
-    }
-
-    commandLine = (char*)malloc(commandLineLength);
-    if (NULL == commandLine)
-    {
-        OsConfigLogError(log, "Cannot run command '%s', cannot allocate %u bytes for command, out of memory", command, (unsigned)commandLineLength);
-        return ENOMEM;
-    }
-
-    snprintf(commandLine, commandLineLength, wrappedCommand ? "%s%s%s%s" : "(%s)%s%s%s", command, commandSeparator, commandTextResultFile, commandTerminator);
-
-    // Execute the command with the requested timeout: error ETIME (62) means the command timed out
-    status = SystemCommand(context, commandLine, timeoutSeconds, callback, log);
+            if (((maxTextResultBytes > 0) && (outputBufferPos == maxTextResultBytes)) || textResult == NULL)
+            {
+                // We don't want any more data, loop to read the rest of the output.
+                continue;
+            }
 
-    free(commandLine);
+            outputBufferSize = bytesRead + outputBufferPos;
+            if ((maxTextResultBytes > 0) && (outputBufferSize > maxTextResultBytes - 1))
+            {
+                outputBufferSize = maxTextResultBytes - 1;
+            }
 
-    // Read the text result from the output of the command, if any, whether command succeeded or failed
-    if (NULL != textResult)
-    {
-        *textResult = NULL;
-        resultsFile = fopen(commandTextResultFile, "r");
-        if (resultsFile)
-        {
-            fseek(resultsFile, 0, SEEK_END);
-            fileSize = ftell(resultsFile);
-            fseek(resultsFile, 0, SEEK_SET);
+            tmp = realloc(*textResult, outputBufferSize + 1);
+            if (NULL == tmp)
+            {
+                OsConfigLogError(log, "Cannot allocate buffer for command '%s'", command);
+                status = ENOMEM;
+                free(*textResult);
+                *textResult = NULL;
+                break;
+            }
+            *textResult = tmp;
 
-            if (fileSize > 0)
+            for (inputBufferPos = 0; (inputBufferPos < bytesRead) && (outputBufferPos < outputBufferSize); inputBufferPos++, outputBufferPos++)
             {
-                // Truncate to desired maximum, if any
-                if ((maxTextResultBytes > 0) && (((size_t)fileSize + 1) > maxTextResultBytes))
+                // Copy the data. Following characters are replaced with spaces:
+                // all special characters from 0x00 to 0x1F except 0x0A (LF) when replaceEol is false
+                // plus 0x22 (") and 0x5C (\) characters that break the JSON envelope when forJson is true
+                const char c = buffer[inputBufferPos];
```

Comment:
>const char c = buffer[inputBufferPos]; [](http://example.com/codeflow?start=16&length=38)

Move local variable definition at beginning of function, C-style. Please correct this everywhere.

In general, can the new variant please more closely follow the original style for names, comments, traces, etc?

Reply:
Thanks (and also I do agree with your points stated here).

Reply:
If you agree (and a quick survey on my team shows that they agree too) then why can't we do it this way? Although It's not a "brace location" style change that is automatable, we can do it gradually in the new and/or modified code, improving the quality and readability. I do understand worries that we won't have consistency in this area, but I think it's still worth it.

Reply:
It appears that declaring local variables at beginning of function is actually C89/ANSI C standard. C99 does not have this requirement. C11 is newer than C99 and also does not have it either. I do agree and my only ask is to do this change gradually, in a way that does not look out of the place within same source code file, component or even per project.